pkgs/development/python-modules/ast-grep-py/default.nix

```diff
@@ -0,0 +1,62 @@
+{
+  lib,
+  buildPythonPackage,
+  rustPlatform,
+  fetchFromGitHub,
+  pytestCheckHook,
+  nix-update-script,
+  ...
+}:
+let
+  inherit (builtins) attrValues;
+  finalAttrs = {
+    pname = "ast-grep-py";
+    version = "0.39.2";
+    pyproject = true;
+
+    src = fetchFromGitHub {
+      owner = "ast-grep";
+      repo = "ast-grep";
+      tag = finalAttrs.version;
+      hash = "sha256-RfsBgxxb9Kd28hzDXNBNFEwpRchxt+VmSMwc2wRDuig=";
+    };
+
+    buildAndTestSubdir = "crates/pyo3";
+
+    cargoDeps = rustPlatform.fetchCargoVendor {
+      inherit (finalAttrs) pname version src;
+      hash = "sha256-5SDGkOeByG8SUQJH/89TLuEJeKcu9lu/ZKbudwCAM0o=";
+    };
+
+    nativeBuildInputs = attrValues {
```

Comment:
I saw you're using `attrValues` and `inherit` rather than `with rustPlatform; [ cargoSetupHook maturinBuildHook ]`, which seems a bit unusual. Unless there's a reason for this, it might be better to stick with the common pattern.